### PR TITLE
For #21443: Disable history delete and search with empty history

### LINKED ISSUE #21443: The delete history button works even when there is no browsing data
### Steps to reproduce

1. Make sure Firefox has no previous browsing data (clean install or delete history)
2. Open the 3-dot menu > History
3. Tap the delete button (trashcan)
4. Select the Delete option from the prompt
5. Observe the snackbar

### Expected behaviour

The button is either hidden or inactive when there is no browsing data to display

### Actual behaviour

The "Browsing data deleted" snackbar is displayed each time the delete button is tapped

### Device name

Samsung Galaxy A8

### Android version

Android 9

### Firefox release type

Firefox Nightly

### Firefox version

94.0.a1 2021-09-22

### Device logs

_No response_

### Additional information

242662878_2089967871180432_482069448327530717_n

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-17974)

### PR DESCRIPTION
**Pull request by fice-t**
This defaults `isEmpty` to true as otherwise entering the history
fragment with no history did not disable the delete button until
another action was taken (I believe this is due to the `HistoryAdapter` behaviour as described below).

Note that due to the way `HistoryAdapter` determines the current empty state, the icons briefly show up as disabled before being enabled when there is history available. That's just cosmetic and could be part of a separate issue (which I can create if necessary).

Alternatively, if anyone knows of an immediate way to get if there is no history available rather than using `historyStore.state.isEmpty` then this can be worked around nicely enough.

https://user-images.githubusercontent.com/13156601/196357315-6e33b484-3d5b-4c6f-90db-9fe4c6747f92.mp4

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Fixes #21443

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt [hunk 1]
```diff
@@ -146,8 +147,8 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
             }
         }.toSet()
 
-    private fun invalidateOptionsMenu() {
-        activity?.invalidateOptionsMenu()
+    private fun invalidateMenu() {
+        activity?.invalidateMenu()
```
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt:151]: I think we can revert this name change in order to keep the diff minimal - `invalidateOptionsMenu` seems fine to me

File: app/src/main/java/org/mozilla/fenix/library/history/HistoryController.kt [hunk 2]
```diff
@@ -64,7 +64,7 @@ class DefaultHistoryController(
     private val openToBrowser: (item: History.Regular) -> Unit,
     private val displayDeleteTimeRange: () -> Unit,
     private val onTimeFrameDeleted: () -> Unit,
-    private val invalidateOptionsMenu: () -> Unit,
+    private val invalidateMenu: () -> Unit,
```
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/library/history/HistoryController.kt:67]: This parameter can keep the old name as well

File: app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt [hunk 3]
```diff
@@ -230,6 +231,15 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
         }
     }
 
+    override fun onPrepareMenu(menu: Menu) {
+        super.onPrepareMenu(menu)
+
+        if (historyStore.state.isEmpty) {
+            menu.findItem(R.id.history_search)?.isEnabled = false
+            menu.findItem(R.id.history_delete)?.isEnabled = false
+        }
```
Comment (MatthewTighe) [inline app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt:240]: I was able to replicate the desired behavior here by moving this into `onCreateMenu`, and we could avoid this additional override.  This also allowed me to switch the `isEmpty` argument to `HistoryState` back to `false`. I assume that the dispatch and call to `invalidateMenu`() in `onEmptyStateChanged` is what is causing the correct behavior

**Review by MatthewTighe — CHANGES_REQUESTED** [review]

**Comment by MatthewTighe** [comment]
Thanks for your contribution! Had a few small suggestions for changes 🙂

**Comment by boek** [comment]
Moved original issue to Bugzilla https://bugzilla.mozilla.org/show_bug.cgi?id=1814192

**Comment by MatthewTighe** [comment]
Hey @fice-t, thanks so much for your contribution. As I mentioned in https://github.com/mozilla-mobile/fenix/pull/27476, we will be moving this repository to [firefox-android](https://github.com/mozilla-mobile/firefox-android) in the coming days, and won't be able review or land this PR until after the migration.

Please feel free to reopen a new PR against the new repo once migration is complete. You will have to clone the new repository and reapply your changes. Thank you again, and sorry for the inconvenience.

It looks like there were some changes requested when this PR was first opened. Please feel free to address those when and if you look to open a PR in the new repo 🙂 

For more information, please see https://github.com/mozilla-mobile/fenix/issues/26855.

**Pull request closed without merging**